### PR TITLE
doc: deprecate the cache tiering

### PR DESCRIPTION
**Pull request by rzarzynski**
This topic has been discussed many times; recently at the Dev Summit of Cephalocon 2023.

This commit is the minial version of the work, contained entirely within the `doc`. However, likely it will be expanded as there were ideas like e.g. adding cache tiering back experimental feature list (Sam) to warn users when deploying a new cluster.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/rados/operations/cache-tiering.rst [hunk 1]
```diff
@@ -1,6 +1,9 @@
 ===============
  Cache Tiering
 ===============
+.. warning:: Cache tiering has been deprecated in the Reef release as it
+             lacks a maintainer for a very long time. This does not mean
```
Comment (anthonyeleven) [inline doc/rados/operations/cache-tiering.rst:5]: s/lacks/has lacked/
below:  s/for that/to do so/

**Review by anthonyeleven — APPROVED** [review]
I've made some grammar suggestions, but those don't warrant blocking.

File: doc/rados/operations/cache-tiering.rst [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
+             it will be certainly removed but the right for that is claimed.
```
Comment (athanatos) [inline doc/rados/operations/cache-tiering.rst:6]: , but we may choose to remove it without much further notice.

File: doc/rados/operations/cache-tiering.rst (hunk 1 above)
Comment (rzarzynski) [reply doc/rados/operations/cache-tiering.rst]: Fixed.

File: doc/rados/operations/cache-tiering.rst (hunk 2 above)
Comment (rzarzynski) [reply doc/rados/operations/cache-tiering.rst]: Changed.

File: doc/rados/operations/cache-tiering.rst [hunk 3]
```diff
@@ -1,6 +1,10 @@
 ===============
  Cache Tiering
 ===============
+.. warning:: Cache tiering has been deprecated in the Reef release as it
+             has lacked a maintainer for a very long time. This does not mean
+             it will be certainly removed, but we may choose to remove it
+             without much further notice
```
Comment (neha-ojha) [inline doc/rados/operations/cache-tiering.rst:7]: nit: missing full stop

**Review by neha-ojha — APPROVED** [review]
@zdover23 please make note of this PR and let's make sure other occurrences of cache tiering in the docs reflect the same message

File: doc/rados/operations/cache-tiering.rst (hunk 3 above)
Comment (rzarzynski) [reply doc/rados/operations/cache-tiering.rst]: Fixed.

**Comment by anthonyeleven** [comment]
> @zdover23 please make note of this PR and let's make sure other occurrences of cache tiering in the docs reflect the same message

I think I worked in some unofficial guidance last year to that end.

**Comment by zdover23** [comment]
@rzarzynski: Is this ready for merge? (And if so, is there any reason to backport it to any feature branches other than Reef?)

**Comment by anthonyeleven** [comment]
I might suggest that backporting has value in this case if it helps someone running an older release from making the mistake of deploying cache tiering.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/51652 - Reef backport
https://github.com/ceph/ceph/pull/51653 - Quincy backport
https://github.com/ceph/ceph/pull/51654 - Pacific backport

**Pull request merged**